### ISSUE TITLE
Force queries to use transactions without explicitly passing one

### ISSUE THREAD

**Issue by Kuzmin** [sequelize/sequelize#2451]
It's pretty useful to write tests inside transactions as it's faster than a full truncate before each test. Therefore i think there should be some option to open a transaction and force all queries through the transaction. A "standard test" right now might look like this;

```
beforeEach(function (done) { 
  sequelize.transaction({ global: true }, function (t) {
    this.transaction = t;
    done();
  }.bind(this));
});

afterEach(function () {
  this.transaction.rollback(function () {
    this.transaction = null;
    done();
  }.bind(this));
});

it('does stuff inside a transaction', function () {
  var user = Models.User.build({ name: 'Alexander' });

  user.save({ transaction: t }).then(function () {
    expect(user.id).to.not.equal(null);

    return user.destroy({ transaction: t });
  }).then(function () {
    return Models.User.findAll({ transaction: t });
  }).then(function (results) {
    expect(results.length).to.equal(0);
  })
});
```

And I think that it's not hard to see how an easily forgotten `{ transaction: t }` might lead to problems. Of course, this example is a bit ridiculous since it's really not a problem, just a bit uncomfortable.

A "real problem" might occur when testing REST API's, where you don't really have direct access to the DB, and thus can't pass any `{ transaction: t }` to the queries. This forces one to use the truncate-all-tables-approach which slows tests down quite a bit when you have 20-something tables and they all reset on each test.. Running a 3 minute test suite on my project, knowing that my last MongoDB-base project took about 20s for a comparable amount of tests is a bit irritating.

Tl:dr; Adding some kind of global transactions would make code utilising Sequelize easier to test, and would also improve running time for tests. Would you consider adding this to roadmap?
